### ISSUE TITLE
launching the emulator without X server runnning

### ISSUE THREAD

**Issue by bert003** [ColinPitrat/caprice32#23]
Until the recent update (about the support for the plus versions), I was able to run the emulator without the X server running.

The emulator would load and display as the below. (Note the error at the top).

![error2](https://cloud.githubusercontent.com/assets/20399702/17082300/eac6c252-5178-11e6-90c5-e2e8b54a567e.jpg)

This did not happen with previous version. It would quit with a message that no display was found (along those lines).

Kindly advise.

**Comment by ColinPitrat** [comment]
This feature is not one of caprice but of SDL. It's able to use aalib (monochrome) or libcaca (colorful - as its name for french people) to display graphics. This can be controlled by SDL_VIDEODRIVER.

I don't know what changed but I expect it to be with your configuration. Maybe this variable was set and is not anymore ? Or libcaca is not installed ?
You can try setting manually the environment variable SDL_VIDEODRIVER=caca
If it doesn't work, check if the library is installed.

On my side I'll improve the error message when video init fails.

**Comment by bert003** [comment]
I just checked.

libcaca is installed and is already the newest version.

I did not change anything in the configuration. I just compiled a fresh clone and accidentally ran cap32 from tty1. I did not update the SDL libraries either.

The thing is that in the previous versions it used to quit when there was no display present. Now it loads with crazy characters and I cannot even quit it since F10 does not work. Can it be set not to run unless we are running it in X?

**Comment by ColinPitrat** [comment]
Oh OK, so the screenshot is what you get now and before it was just exiting with an error message ?
It's actually a nice feature, even though the fact that Fx keys not working is painful.
You can try a game in this mode you'll see. You have to type characters blindly to start it but then the game should be OK.

Unfortunately, the only way to quit if you don't have a joystick is to kill it from another terminal as Control-Z doesn't work. I could add an alternative key for the menu just for this if I can find a free one :-)

**Comment by bert003** [comment]
Yes the screenshot is what I get now. Before it was just exiting with a message. Honestly I prefer it like that.

Can it be set as before or was this left on purpose? Since, to be honest, it is not very user-friendly you know :-)

**Comment by ColinPitrat** [comment]
I'm not sure if there's a way to prevent SDL from using libcaca from the code. I'll have a look.

But even if it's possible I would make it an option because I definitely find it cool to be able to use caprice in console (for now with joysticks and yes, you have to now what's supposed to be on the screen when using the menu or typing stuff, and not only F1->F10 are missing but also shift, but still ...).

**Comment by bert003** [comment]
Yes I find it cool as well to be able to launch caprice32 from the console ONLY if the screen was legible... Otherwise things could get confusing to the normal user...

**Comment by cpcitor** [comment]
It would be really fun if libcaca happened to locally analyze each n*m color patch to find the most resembling character*bgcolor*fgcolor combination and display that character*.
That would need some "grid detection".

Then running in a text console would actually produce something readable! :stuck_out_tongue: 


*CPC firmware does character recognition already, entry TXT RD CHAR and indirection TXT UNWRITE, though only if bg and bg colors are known.

**Comment by ColinPitrat** [comment]
> It would be really fun if libcaca happened to locally analyze each n*m color patch to find the most resembling characterbgcolorfgcolor combination and display that character[*].

That's what it does but a character in the console doesn't correspond to a character of the CPC nor are they necessarily aligned. With a high resolution terminal the result is actually not too bad.
